### PR TITLE
helper & database scripts cleaning

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
This PR is about cleaning our helper script from multi-database support without changing functional signatures.
shellcheck -s sh has been used to sanitize the code.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [ ] PR name is meaningful
- [ ] Purpose of the PR is understood
- [ ] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [ ] Commit descriptions are understandable and well formatted

##### Source code quality

- [ ] Source code changes are understandable
- [ ] Variable and function names are meaningful
- [ ] Code comments are correct (logically and spelling)
- [ ] Required documentation changes are present and part of the PR
- [ ] `bareos-check-sources --since-merge` does not report any problems
- [ ] `git status` should not report modifications in the source tree after building and testing

##### Tests

- [ ] Decision taken that a system- or unittest is required (if not, then remove this paragraph)
- [ ] The decision towards a systemtest is reasonable compared to a unittest
- [ ] Testname matches exactly what is being tested
- [ ] Output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: core/src/cats/create_bareos_database.in [hunk 1]
```diff
@@ -31,104 +31,51 @@
...
-         shift
-         ;;
+  case "$1" in
       postgresql)
-         db_type=$1
+         warn "Parameter db_type is no more needed"
          shift
          ;;
       *)
+         error "db_type $1 is not more supported, only postgresql"
```
Comment (pstorz) [inline core/src/cats/create_bareos_database.in:47]:
```suggestion
         error "db_type $1 is no more supported, only postgresql"
```

File: core/src/cats/create_bareos_database.in [hunk 2]
```diff
@@ -31,104 +31,51 @@
...
-default_db_type=`get_database_driver_default`
-working_dir=`get_working_dir`
-
-
-#
-# See if the first argument is a valid backend name.
-# If so the user overrides the default database backend.
-#
+# Check if some parameters are given, try to detect old usage with db_type
 if [ $# -gt 0 ]; then
```
Comment (pstorz) [inline core/src/cats/create_bareos_database.in:40]: Do we want to cancel if we have more than one parameter?

File: core/src/cats/drop_bareos_database.in [hunk 3]
```diff
@@ -30,71 +30,34 @@
...
-db_user="${db_user:-`get_database_user @db_user@`}"
-default_db_type=`get_database_driver_default`
-working_dir=`get_working_dir`
-
-#
-# See if the first argument is a valid backend name.
-# If so the user overrides the default database backend.
-#
+# Check if some parameters are given, try to detect old usage with db_type
 if [ $# -gt 0 ]; then
```
Comment (pstorz) [inline core/src/cats/drop_bareos_database.in:39]: Same as above: Do we want to exit if more than one parameter is set?

File: core/src/cats/drop_bareos_tables.in [hunk 4]
```diff
@@ -31,109 +31,55 @@
...
-         shift
-         ;;
+  case "$1" in
       postgresql)
-         db_type=$1
+         warn "Parameter db_type is no more needed"
          shift
          ;;
       *)
+         error "db_type $1 is not more supported, only postgresql"
```
Comment (pstorz) [inline core/src/cats/drop_bareos_tables.in:49]: The above lines are the same in all  scripts as it seems, don't we want to put them into one function in the lib?

**Review by pstorz — CHANGES_REQUESTED** [review]
Good work! Some small typos.

I think there is potential to refactor and put common parts into a common function into the library.

File: core/src/cats/drop_bareos_tables.in (hunk 4 above)
Comment (bruno-at-bareos) [reply core/src/cats/drop_bareos_tables.in:49]: That's seems to be a good idea, They were there and I forgot to imagine this, my first idea was to simply remove all, and don't be nice with older calls.

File: core/src/cats/create_bareos_database.in (hunk 1 above)
Comment (bruno-at-bareos) [reply core/src/cats/create_bareos_database.in]: included in next push

File: core/src/cats/drop_bareos_database.in (hunk 3 above)
Comment (bruno-at-bareos) [reply core/src/cats/drop_bareos_database.in]: No we will not exit, but we won't take care of any, nor pass all to database utility.

**Comment by bruno-at-bareos** [comment]
Return to draft status, we need to get rid of dbcheck dependency in bareos-config-lib.sh

**Comment by bruno-at-bareos** [comment]
This request is closed and integrated directly to PR#1081

**Pull request closed without merging**